mir_eval/separation.py

```diff
@@ -161,28 +166,139 @@ def bss_eval_sources(reference_sources, estimated_sources):
+    reference_sources,
+    estimated_sources,
+    win,
+    hop,
+    compute_permutation=False
+):
+    """Framewise computation of bss_eval_sources
+
+    Examples
+    --------
+    >>> # reference_sources[n] should be an ndarray of samples of the
+    >>> # n'th reference source
+    >>> # estimated_sources[n] should be the same for the n'th estimated
+    >>> # source
+    >>> (sdr, sir, sar,
+    ...  perm) = mir_eval.separation.bss_eval_sources_framewise(
+             reference_sources,
+    ...      estimated_sources)
+
+    Parameters
+    ----------
+    reference_sources : np.ndarray, shape=(nsrc, nsampl)
+        matrix containing true sources
+    estimated_sources : np.ndarray, shape=(nsrc, nsampl)
+        matrix containing estimated sources
+    win : int, optional
+        Window length for framewise evaluation
+    hop : int, optional
+        Hop size for framewise evaluation
+    compute_permutation : bool, optional
+        compute permutation of estimate/source combinations for all windows
+        (False by default)
+
+    Returns
+    -------
+    sdr : np.ndarray, shape=(nsrc, nframes)
+        vector of Signal to Distortion Ratios (SDR)
+    sir : np.ndarray, shape=(nsrc, nframes)
+        vector of Source to Interference Ratios (SIR)
+    sar : np.ndarray, shape=(nsrc, nframes)
+        vector of Sources to Artifacts Ratios (SAR)
+    perm : np.ndarray, shape=(nsrc, nframes)
+        vector containing the best ordering of estimated sources in
+        the mean SIR sense (estimated source number perm[j] corresponds to
+        true source number j)
+        Note: perm will be range(nsrc) for all windows if compute_permutation
+        is False
+
+    """
+
+    # make sure the input is of shape (nsrc, nsampl)
+    if estimated_sources.ndim == 1:
+        estimated_sources = estimated_sources[np.newaxis, :]
+    if reference_sources.ndim == 1:
+        reference_sources = reference_sources[np.newaxis, :]
+
+    validate(reference_sources, estimated_sources)
+    # If empty matrices were supplied, return empty lists (special case)
+    if reference_sources.size == 0 or estimated_sources.size == 0:
+        return np.array([]), np.array([]), np.array([]), np.array([])
+
+    nsrc = reference_sources.shape[0]
+
+    nwin = int(
+        np.floor((reference_sources.shape[1] - win + hop) / hop)
+    )
+    # make sure that more than 1 window will be evaluated
+    if nwin < 2:
+        raise ValueError('Invalid window size and hop size have been supplied.'
+                         'From these paramters it was determined that {} '
+                         'windows should be used.'.format(nwin))
+
+    # compute the criteria across all windows
+    SDR = np.empty((nsrc, nwin))
+    SIR = np.empty((nsrc, nwin))
+    SAR = np.empty((nsrc, nwin))
+    perm = np.empty((nsrc, nwin))
+
+    for k in range(nwin):
+        K = slice(k * hop, k * hop + win)
+        SDR[:, k], SIR[:, k], SAR[:, k], perm[:, k] = bss_eval_sources(
```

Comment:
It would be better to use a different variable name here.  `K` and `k` are too similar, and neither is descriptive.